capstone/src/test/java/com/google/sps/servlets/CheckNewUserServletTest.java

```diff
@@ -0,0 +1,136 @@
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.ArrayList;
+
+/**
+ * Unit tests for {@link CheckNewUserServlet}.
+ */
+ @RunWith(JUnit4.class)
+public class CheckNewUserServletTest {
+  private static final String CURRENT_USER_EMAIL = "test@mctest.com";
+  private static final String CURRENT_USER_ID = "testy-mc-test";
+  private static final String OTHER_USER_EMAIL = "test2@mctest.com";
+  private static final String OTHER_USER_ID = "testytwo-mc-test";
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+            new LocalDatastoreServiceTestConfig()
+                .setDefaultHighRepJobPolicyUnappliedJobPercentage(0),
+            new LocalUserServiceTestConfig())
+          .setEnvEmail(CURRENT_USER_EMAIL)
+          .setEnvIsLoggedIn(true)
+          .setEnvAuthDomain("gmail.com")
+          .setEnvAttributes(
+              new HashMap(
+                  ImmutableMap.of(
+                      "com.google.appengine.api.users.UserService.user_id_key", CURRENT_USER_ID)));
+
+  private static final User CURRENT_USER = new User(CURRENT_USER_ID, "Test", "McTest", 
+                          CURRENT_USER_EMAIL, 
+                          /* phoneNumber= */ "123-456-7890", 
+                          /* profilePic= */ "", 
+                          /* badges= */ new LinkedHashSet<Badge>(), 
+                          /* groups= */ new LinkedHashSet<Long>(), 
+                          /* interests= */ new ArrayList<String>());
+
+  private static final User OTHER_USER = new User(OTHER_USER_ID, "Test Two", "McTest", 
+                          OTHER_USER_EMAIL, 
+                          /* phoneNumber= */ "123-456-7890", 
+                          /* profilePic= */ "", 
+                          /* badges= */ new LinkedHashSet<Badge>(), 
+                          /* groups= */ new LinkedHashSet<Long>(), 
+                          /* interests= */ new ArrayList<String>());                        
+
+  @Mock private HttpServletRequest mockRequest;
+  @Mock private HttpServletResponse mockResponse;
+  private StringWriter responseWriter;
+  private DatastoreService datastore;
+  private CheckNewUserServlet checkNewUserServlet;
+
+  @Before
+  public void setUp() throws Exception {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+
+    addUserToDatastore(datastore, OTHER_USER);
+
+    // Set up a fake HTTP response.
+    responseWriter = new StringWriter();
+    when(mockResponse.getWriter()).thenReturn(new PrintWriter(responseWriter));
+
+    checkNewUserServlet = new CheckNewUserServlet();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
```

Comment:
You should set your other instance variables to null (responseWriter, datastore, checkNewUserServlet)

Reply:
I'm not sure I follow, do you just mean instead of

`private StringWriter responseWriter;`

she should have

`private StringWriter responseWriter = null;`?

I actually think the approach of **not** setting it null makes the tests fail much more likely if a new variable was added and you forgot to set it up for a test in `setUp` or `tearDown`, which I would argue is a good thing that reminds the developer to add a new variable to the `setUp`/`tearDown`

The only note I would have is to make sure that any variable that you're declaring up above should either be final or explicitly updated in `setUp` and/or `tearDown` to ensure you don't accidentally leak data between tests, which you appear to be doing.

Reply:
I meant that in tearDown she should have `responseWriter = null;`